stdlib/mexpr/eq.mc

```diff
@@ -362,14 +362,47 @@ lang VarPatEq = VarPat
     else None ()
 end
 
-lang SeqTotPatEq
+lang SeqTotPatEq = SeqTotPat
   sem eqPat (env : Env) (free : Env) (patEnv : NameEnv) (lhs : Pat) =
-  -- TODO(dlunde,2020-09-29)
+  | PSeqTot {pats = ps2} ->
+    match lhs with PSeqTot {pats = ps1} then
+      if eqi (length ps2) (length ps1) then
+        let z = zipWith (lam p1. lam p2. (p1,p2)) ps1 ps2 in
+        optionFoldlM (lam fpEnv. lam ps.
+          match fpEnv with (f,p) then
+            eqPat env f p ps.0 ps.1 
+          else never)
+          (free,patEnv) z
+      else None ()
+    else None ()
 end
 
-lang SeqEdgPatEq
+lang SeqEdgPatEq = SeqEdgePat
+  -- NOTE(gcaylak,2020-10-02): 'SeqEdgPatEq' is called 'SeqEdgePat' in 'ast.mc'.
+  -- Should we change 'SeqEdgPatEq' to 'SeqEdgePatEq' as well?
   sem eqPat (env : Env) (free : Env) (patEnv : NameEnv) (lhs : Pat) =
-  -- TODO(dlunde,2020-09-29)
+  | PSeqEdge {prefix = pre2, middle = mid2, postfix = post2} ->
+    match lhs with PSeqEdge {prefix = pre1, middle = mid1, postfix = post1} then
+      match _eqpatname patEnv free mid1 mid2 with Some (f,p) then
+        if eqi (length pre1) (length pre2) then
+          let z1 = zipWith (lam p1. lam p2. (p1,p2)) pre1 pre2 in
+          let z2 = zipWith (lam p1. lam p2. (p1,p2)) post1 post2 in
+          match optionFoldlM (lam fpEnv. lam ps.
```

Comment:
I'd prefer binding the match target (starting with `optionFoldlM (lam fpEnv.`) to a variable before the match. It is difficult to read at the moment.